python-main.js

```diff
@@ -337,6 +337,7 @@ function web_editor(config) {
     // Loads Python code into the editor and/or filesystem
     function loadPy(filename, codeStr) {
         var isModule = isPyModule(codeStr);
+        var filesystemName = filename;
```

Comment:
They way you are using this variable it's the same as the one created below this
```js
var moduleName = filename.replace('.py', '');
```
So no need to create this one, you can just reuse that one.

Reply:
The line above this sets filename to 'main.py' if the file that is being loaded isn't a module. I added this line to retain the original filename. Otherwise, we get the behaviour described here: https://github.com/bbcmicrobit/PythonEditor/issues/108#issuecomment-470492052

Reply:
Then `var moduleName = filename.replace('.py', '')` should be moved before that line. 
This variable is only used to display the module name in alerts and confirms, so it should be safe to move. Do still check if what I just said is true.

Reply:
LGTM 👍